### PR TITLE
Setup Notebook Testing

### PR DESCRIPTION
**Pull request by christian512**
Introducing Markdown Notebooks (generated with [JuPyText](https://jupytext.readthedocs.io/en/latest/)) and an automatic testing of notebooks using [nbmake](https://github.com/treebeardtech/nbmake) in GitHub Actions. The executed notebooks ares stored as Artifacts by the GitHub Action.

### CODE REVIEW

File: test_environment.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Qutip-Test-Env
```
Comment (hodgestar) [inline test_environment.yml:1]:
```suggestion
name: test-environment
```

File: test_environment.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+channels:
+- conda-forge
+dependencies:
+- python==3.7.3
```
Comment (hodgestar) [inline test_environment.yml:5]: It looks like these requirements were taken from the qutip-notebook repository, which is fine, but the versions are very old. Let's leave them as is for this PR but open an issue to update the versions to more recent ones.

File: test_environment.yml (hunk 1 above)
Comment (hodgestar) [reply test_environment.yml:1]: The capitalized Qutip-Test-Env just looks odd to me and makes it sound like this is an environment for testing QuTiP itself. :)

**Review by hodgestar — APPROVED** [review]
Thanks @christian512. The changes look good.

I would like us to create folders for the notebooks under tutorials as soon as possible, but happy for that to happen in a follow up PR. I'm not sure what the folders should be, but they should group the notebooks into suitable categories and we shouldn't have any notebooks lying outside in the top-level `tutorials` folder.

File: test_environment.yml (hunk 2 above)
Comment (christian512) [reply test_environment.yml:5]: Alright, I will open an issue after the PR is merged.

**Pull request merged**